src/rpc/mod.rs

```diff
@@ -180,8 +180,15 @@ impl ConnectionSender {
         self.sender.send(message).await.map_err(Self::broken_pipe)
     }
 
+    #[allow(dead_code)]
+    /// close the connection
     async fn close(&mut self) -> Result<()> {
         self.request(ObTablePacket::ClosePoison).await?;
+        self.shutdown()
```

Comment:
If we want to close the ConnectionSender gracefully, I guess we should join the `writer` handle after sending the poison signal.